model/execution/DeliveryExecutionList.php

```diff
@@ -161,7 +159,16 @@ private function createExecution($cachedData, $extraFields): array
         return $execution;
     }
 
-    private function getLastPauseReason(string $deliveryExecutionId): ?array
+    private function isPausedByProctor($lastPause): bool
+    {
+        return isset(
+            $lastPause[0][DeliveryLog::DATA]['reason'],
+            $lastPause[0][DeliveryLog::DATA]['context']
+        )
+        && $lastPause[0][DeliveryLog::DATA]['context'] === '/taoProctoring/Monitor/pauseExecutions';
```

Comment:
It works like this. But I think it would be better to use `created_by` column from delivery_log to check if last pause event was done by Proctor. It looks more obvious to use user  role instead of controller which was called.

Reply:
Sorry @SashaTAO , I can't agree with you here. Now we can't divide users who launches Proctor session from the users who launches delivery execution in the test runner. So it will be not so obvious why do I see or not see predefined reasons. For example : if I use the same user to run test and to see proctors section: I will see everything predefined, even systems messages that `System generated pause`. And to other users and for me in a week it will be confusing again to understand what is not correct here, why do I see such messages.
I am categorically against the use of roles, to display the reasons, because I'm see here issues: for QA, for customers, for developers who forget or didn't know that behavior (so it is for everyone).

I already suggested you other solutions:
1. More clear as for me: I can add into the reasons new parameter (triggered on the proctor screen or something that means that), then I can use only these messages for the lastPause pre-populated reasons.
2. We can try to fix roles detection (it will be really complex solution, but we can determine that the user who has role student (test taker) - from the test runner should be marked as test taker, and if this same user makes actions as a proctor, then we can mark these actions as a made by proctor). But again, it is complex solution that takes a lot of time, and I don't think that this feature cost this time.